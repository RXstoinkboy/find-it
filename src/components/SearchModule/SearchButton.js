import React from 'react';
import styles from './styles/SearchButton.module.scss'
import {ReactComponent as SearchIcon} from '../../assets/img/search.svg'
import PropTypes from 'prop-types'

export const SearchButton = props => {
    return (
        <button 
            // type='submit'
            className={styles.searchButton}
            onClick={props.handleGetBusinessesList}
        >
            <SearchIcon className={styles.searchIcon}/>
        </button>
    );
};

export default SearchButton;

SearchButton.propTypes = {
    handleGetBusinessesList: PropTypes.func.isRequired
}