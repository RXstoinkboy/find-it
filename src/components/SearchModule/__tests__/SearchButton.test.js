import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import {SearchButton} from '../SearchButton'

it(`SearchButton renders correctly`, ()=> {
    const mockFn = jest.fn();
    const searchButton = shallow(
        <SearchButton onClick={mockFn} handleGetBusinessesList={mockFn}/>
    )

    searchButton.simulate('click');
    expect(mockFn.mock.calls.length).toBe(1);
    expect(toJson(searchButton)).toMatchSnapshot();
})