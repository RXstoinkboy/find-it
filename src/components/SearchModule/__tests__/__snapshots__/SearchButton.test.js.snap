// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton renders correctly 1`] = `
<button
  className="searchButton"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <ForwardRef
    className="searchIcon"
  />
</button>
`;
